### PR TITLE
Export no arg

### PR DESCRIPTION
**Pull request by lgyger**
we can merge it I guess

### CODE REVIEW

File: srcs/ft_memcpy.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+void	*ft_memcpy(void *dst, const void *src, size_t n)
+{
+	unsigned char		*dst2;
+	const unsigned char	*src2;
+	size_t				i;
+
+	dst2 = (unsigned char *) dst;
+	src2 = (unsigned char *) src;
+	if (dst == NULL && src == NULL && n >= 0)
```
Comment (lgyger) [inline srcs/ft_memcpy.c:23]:
```suggestion
	if (dst == NULL && src == NULL && n != 0)
```

**Review by lgyger — COMMENTED** [review]
resolved the issue of checking if unsigned var is positive

File: srcs/unset.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+/*   By: lgyger <marvin@42lausanne.ch>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/06/18 19:58:43 by lgyger            #+#    #+#             */
+/*   Updated: 2022/06/18 20:49:36 by lgyger           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+static	int	getpos(const char *str, const char c)
```
Comment (lgyger) [inline srcs/unset.c:15]:
```suggestion
static	inline int	getpos(const char *str, const char c)
```

File: srcs/unset.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	{
+		pos = get_index(command->arg[i], *envp);
+		if (pos != -1)
+		{
+			new_env = crt_nenv(*envp, pos);
+			*envp = new_env;
+		}
+		++i;
+	}
+}
```
Comment (lgyger) [inline srcs/unset.c:84]:
```suggestion

	if (!envp[0])
		return ;
	i = 0;
	while (command->arg[++i])
	{
		pos = get_index(command->arg[i], *envp);
		if (pos != -1)
			*envp = crt_nenv(*envp, pos);
	}
}
```

File: srcs/unset.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	i = 0;
+	while (ev[i])
+	{
+		if (!ft_strncmp(tf, ev[i], ft_strlen(tf))
+			&& getpos(ev[i], '=') == ft_strlen(tf))
+			return (i);
+		++i;
+	}
+	return (-1);
+}
```
Comment (lgyger) [inline srcs/unset.c:33]:
```suggestion
static int	get_index(char *tf, char	**ev)
{
	size_t	i;

	i = -1;
	while (ev[++i])
		if (!ft_strncmp(tf, ev[i], ft_strlen(tf))
			&& getpos(ev[i], '=') == ft_strlen(tf))
			return (i);
	return (-1);
}
```

File: srcs/unset.c [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+	i = 0;
+	ret = 0;
+	while (i != pos && ev[i])
+	{
+		ret += sizeof(ev[i]);
+		++i;
+	}
+	return (ret);
+}
```
Comment (lgyger) [inline srcs/unset.c:48]:
```suggestion
size_t	nbr_bytes( char **ev, const int pos)
{
	size_t	ret;
	int		i;

	i = -1;
	ret = 0;
	while (++i != pos && ev[i])
		ret += sizeof(ev[i]);
	return (ret);
}
```

File: srcs/unset.c [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	char	**new_env;
+
+	len = 0;
+	while (ev[len])
+		++len;
+	new_env = malloc(sizeof(char *) * len);
+	new_env[len - 1] = NULL;
+	ft_memcpy(new_env, ev, nbr_bytes(ev, pos));
+	ft_memcpy(new_env + pos, ev + pos + 1, nbr_bytes(ev + pos + 1, -1));
+	return (new_env);
```
Comment (lgyger) [inline srcs/unset.c:62]:
```suggestion
        ++pos;
	ft_memcpy(new_env + pos - 1,  ev + pos, nbr_bytes(ev + pos ,  -1));
	free(ev);
	free(*(ev + pos - 1));
	return (new_env);
```

**Review by lgyger — COMMENTED** [review]
redid what I already have done, I want to kill myself

File: srcs/ft_strncmp.c [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+/* ************************************************************************** */
+#include"minishell.h"
+
+int	ft_strncmp(const char *s1, const char *s2, size_t n)
+{
+	unsigned int	i;
+
+	i = 0;
+	if (n == 0)
+		return (0);
```
Comment (lgyger) [inline srcs/ft_strncmp.c:20]:
```suggestion
{
	if (n == 0)
		return (0);
```

**Review by lgyger — COMMENTED** [review]
removed that dumb line

File: srcs/unset.c [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+	int		i;
+
+	i = -1;
+	ret = 0;
+	while (++i != pos && ev[i])
+		ret += sizeof(ev[i]);
+	return (ret);
+}
+
+char	**crt_nenv(char **ev, const size_t pos)
```
Comment (lgyger) [inline srcs/unset.c:44]:
```suggestion
char	**crt_nenv(char **ev, size_t pos)
```

File: srcs/unset.c [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+{
+	size_t	len;
```
Comment (lgyger) [inline srcs/unset.c:46]:
```suggestion

```

**Review by lgyger — COMMENTED** [review]
thanks you were right I forgot those 2

**Review by ldominiq — APPROVED** [review]

**Pull request merged**